app/queries/proposal_listing_query.rb

```diff
@@ -121,12 +121,20 @@ def apply_text_filter(proposals_data)
 
   def apply_paging_filter(proposals_data)
     unless params[:text] || params[user.client_model_slug.to_sym]
-      limit = params[:size] || params[:limit] || Proposal::MAX_SEARCH_RESULTS
-      offset = params[:from] || 0
+      limit = (params[:size] || params[:limit] || Proposal::MAX_SEARCH_RESULTS).to_i
```

Comment:
why do we accept either `size` or `limit` here? It might be easier to understand code if we only accept one or the other, but maybe there's a good reason why we need to look for either?